File: go-client-sdk/pkg/models/operations/getonecrmlead.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmLeadQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmLeadResponseBody struct {
	Lead *shared.Lead `json:"lead,omitempty"`
}

type GetOneCrmLeadRequest struct {
	QueryParams GetOneCrmLeadQueryParams
}

type GetOneCrmLeadResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmLeadResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmlead.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmLeadQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmLeadResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=lead" })
  lead?: shared.Lead;
}


export class GetOneCrmLeadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmLeadQueryParams;
}


export class GetOneCrmLeadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmLeadResponseBody;
}
